hal/ia32/string.c

```diff
@@ -54,30 +55,31 @@ int hal_memcmp(const void *ptr1, const void *ptr2, size_t num)
 
 void hal_memset(void *dst, int v, size_t l)
 {
-	__asm__ volatile
-	(" \
-		cld; \
-		movl %0, %%ecx; \
-		movl %%ecx, %%edx; \
-		andl $3, %%edx; \
-		shrl $2, %%ecx; \
-		\
-		xorl %%eax, %%eax; \
-		movb %1, %%al; \
-		movl %%eax, %%ebx; \
-		shll $8, %%ebx; \
-		orl %%ebx, %%eax; \
-		movl %%eax, %%ebx; \
-		shll $16, %%ebx; \
-		orl %%ebx, %%eax; \
-		\
-		movl %2, %%edi; \
-		rep; stosl; \
-		movl %%edx, %%ecx; \
-		rep; stosb"
+	/* clang-format off */
+	__asm__ volatile (
+		"cld\n\t"
+		"movl %0, %%ecx\n\t"
+		"movl %%ecx, %%edx\n\t"
+		"andl $3, %%edx\n\t"
```

Comment:
Shorter
```suggestion
		"andb $3, %%dl\n\t"
```

Reply:
Wrong suggestion.
For example:
```
movl $0x100, %%ecx
movl %%ecx, %%edx
andb $3, %%dl
```
`edx` register will have a value of `0x100` instead `0`.

Reply:
Ack. I marked conversation as resolved, wrong suggestion not be used. Thanks for catching this.